posts/2021-01-22-ee9-messaging-security-21002-beta.adoc

```diff
@@ -0,0 +1,205 @@
+* Jakarta Authentication 2.0 (`jaspic-2.0`)
+* Jakarta Authorization 2.0 (`jacc-2.0`)
+* Jakarta Persistence 3.0 (includes Eclipselink 3.0-RC1.) (`jpa-3.0`)
+* Jakarta XML Binding 3.0 (`jaxb-3.0`)
+* Jakarta Managed Beans 2.0 (`managedBeans-2.0`)
+* Jakarta Concurrency 2.0 (`concurrent-2.0`)
+* Jakarta Bean Validation 3.0 (`beanValidation-3.0`)
+* Jakarta Contexts and Dependency Injection 3.0 (`cdi-3.0`)
+* Message-Driven Beans 4.0 (`mdb-4.0`)
+* JDBC 4.2 & 4.3 (`jdbc-4.2` & `jdbc-4.3`)
+* Jakarta Transactions 2.0 (`transaction-2.0`)
+* Jakarta JSON Binding 2.0 (`jsonb-2.0`)
+* Jakarta JSON Processing 2.0 (`jsonp-2.0`)
+* Jakarta Servlet 5.0 (`servlet-5.0`)
+* Jakarta Server Pages 3.0 (`jsp-3.0` now `pages-3.0`)
+* Jakarta Expression Language 4.0 (`el-4.0` now `expressionLanguage-4.0`)
+
+[#messaging]
+=== Messaging 3.0
+
+Jakarta Messaging is an API and services that enable applications to create, send, and receive messages via loosely coupled, reliable asynchronous communications. Support for version 3.0 updates the API and services to the jakarta.* namespace and introduces four new features for Jakarta EE 9: `messaging-3.0`, `messagingClient-3.0`, `messagingServer-3.0`, `messagingSecurity-3.0`.
+
+Configuration is the same as the corresponding features for Jakarta EE 8 (`jms-2.0`, `wasJmsClient-2.0`, `wasJmsServer-1.0`, `wasJmsSecurity-1.0`) with one exception: version 3.0 support does not introduce a new feature corresponding to `jmsMdb-1.0`. Instead, use Jakarta EE 9 feature `mdb-4.0` when deploying message-driven beans to connect to MQ providers, JCA resource adapters, or the embedded messaging server.
+
+To enable the new Jakarta EE 9 messaging features, add them to your `server.xml`. Here's a feature configuration that enables applications to access message queues via the Messaging 3.0 API, enables message-driven beans to connect to the embedded messaging server, and enables JNDI lookups for messaging resources.
+
+The following features can be included in your server.xml:
+
+[source, xml]
+----
+ <featureManager>
+       <feature>messagingClient-3.0</feature>
+       <feature>mdb-4.0</feature>
+       <feature>jndi-1.0</feature>
+ </featureManager>
+----
+
+Additional information about the Jakarta Server Faces 3.0 specification can be found link:https://jakarta.ee/specifications/faces/3.0/[here].
+
+[#security]
+=== Security 2.0
+
+The new `appSecurity-4.0` feature has been created to support Jakarta EE 9 and introduces Jakarta Security 2.0 support. The `appSecurityClient-1.0`, `audit-1.0`, `constrainedDelegation-1.0`, `jcaInboundSecurity-1.0`, `jwt-1.0`, `passwordUtilities-1.0` and `spnego-1.0` features have been updated to also support Jakarta EE 9.
+
+The appSecurity-4.0 feature must be added to `server.xml` when using application security with Jakarta EE 9. The other `security-2.0` will automatically adapt to the level of Java EE or Jakarta EE that is already in use, so no change is needed when using them with Jakarta EE 9.
```

Comment:
The other `security-2.0` ***features*** will ...

Reply:
Decided to go with `The other security-2.0 features will`

Reply:
The other features are NOT Jakarta features which is why it confuses me to use that the "other security-2.0 features" terminology.  It is also weird because security-2.0 is not a liberty feature name.

Reply:
Ah I see now, thanks. `The other security-2.0 features will` -> `The other Liberty security features will`